daemon/src/main/java/org/lsposed/lspd/service/LSPManagerService.java

```diff
@@ -263,6 +263,12 @@ synchronized boolean preStartManager(String pkgName, Intent intent) {
                     // or it's run by ourself after killing, resume it
                     return true;
                 } else {
+                    // Check the flag in case new launch comes before finishing
+                    // the previous one to avoid racing.
+                    if (pendingManager) {
+                        Log.d(TAG, "manager is still on its way when new launch comes, skipping");
+                        return false;
+                    }
```

Comment:
```suggestion
                } else if (pendingManager) {
                    // Check the flag in case new launch comes before finishing
                    // the previous one to avoid racing.
                    Log.d(TAG, "manager is still on its way when new launch comes, skipping");
                    return false;
                } else {
```